### ISSUE TITLE
openHAB MQTT Channel Autodiscovery not working cause name=null

### ISSUE THREAD

**Issue by BOFH90** [gysmo38/mitsubishi2MQTT#245]
After flashing and connecting to WiFi and setting up the MQTT-connection, openHAB discovered the device, but showed no available channels. After looking through the openhab.log, I discovered that the HA-config could not be adopted, cause the "name" in the config-topic was "null". So I edited the config and changed the name from nullptr to something else, the channels were successfully created. 
 
That's the part, that needs to be adjusted.:
`haConfig["name"]                          = nullptr;`

Also the swing-modes are not detected by openHAB. I have to find a solution to this problem as well.

**Comment by willyfromtheblock** [comment]
https://www.openhab.org/addons/bindings/mqtt.homeassistant/

_The HomeAssistant Climate Components is not yet supported._

Are you certain this will work even if this field are set correctly?

**Comment by BOFH90** [comment]
Yes, I do. I edited the haConfig to something random for my devices and the OH Autodiscovery worked after that.
